docs/gallery/using_nisq_circuits.ipynb

```diff
@@ -0,0 +1,530 @@
+    "\n",
+    "# Import QASM into PyZX\n",
+    "zx_circuit = zx.Circuit.from_qasm(qasm_str)\n",
+    "\n",
+    "# Convert incoming circuit into graph\n",
+    "zx_graph = zx_circuit.to_graph()\n",
+    "\n",
+    "# Draw graph of incoming circuit\n",
+    "zx.draw(zx_graph, labels = True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "de2c9b1d",
+   "metadata": {},
+   "source": [
+    "The incoming circuit is not optimised. Topologiq can handle an unoptimised circuit, but the space-time volume of the diagram produced by topologiq is increasingly proportional to the size of the input graph. It is therefore a good idea to optimise the incorming circuit. \n",
+    "\n",
+    "For this, since PyZX's QASM compatibility is still in-development, we need to start by initialising the qubits after converting the circuit into a graph and adding measurement bases for the ancilla qubits."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ad835257",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Apply states\n",
+    "# Credits. C, Purva Thakre\n",
+    "num_apply_state = zx_graph.num_inputs()\n",
+    "zx_graph.apply_state('0' * num_apply_state)\n",
+    "\n",
+    "# Apply post-select only to the outputs of the ancilla qubits\n",
+    "# Credits. C, Purva Thakre\n",
+    "zx_graph.apply_effect('000///////')\n",
+    "zx.draw(zx_graph, labels = True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7deecf1c",
+   "metadata": {},
+   "source": [
+    "It is now possible to use more-standard PyZX methods to optimise the circuit/graph further."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c5ed1cc6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# From Aleks Kissinger's notebook: https://nbviewer.org/github/zxcalc/pyzx/blob/master/demos/example-circuit-simp.ipynb\n",
+    "zx.full_reduce(zx_graph)\n",
+    "zx.to_rg(zx_graph)\n",
+    "\n",
+    "random.seed(12)\n",
+    "zx.draw(zx_graph, labels = True, auto_layout=True)\n",
+    "\n",
+    "# Object below is not typical in a PyZX workflow but useful for visualisations in the next section.\n",
+    "fig_data = zx.draw_matplotlib(zx_graph, labels=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "25cd1c27",
+   "metadata": {},
+   "source": [
+    "## Algorithmic lattice surgery"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "38775219",
+   "metadata": {},
+   "source": [
+    "It is now possible to convert this graph into `block_graph` compatible objects by giving it to topologiq. \n",
+    "\n",
+    "The first is to convert the graph into topologiq's native graph object, a `simple_graph`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8914774a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from topologiq.scripts.runner import runner\n",
+    "from topologiq.utils.interop_pyzx import pyzx_g_to_simple_g\n",
```

Comment:
`build-and-test-python` workflows are failing because `uv` cannot find the `topologiq` installation. Looks like topologiq should be a tests dependency as well as docs dependency in the toml file.